### PR TITLE
Introduce `consentHandlingOverride` for amp-ad

### LINKED ISSUE #15269: Introduce `consentHandlingOverride` for amp-ad
For a page that has amp-consent configured to block an amp-ad: 

Current behavior:
- amp-ad is not scheduled until user interacted (accept/reject/dismiss) with the consent prompt.
- the amp-ad 3p iframe is then built and the consent state is made available to the 3p js
- it's 3p ad networks' responsibility to respect the consent state and do the right thing (NPA or no ads)

New behavior
- introduce a new field in ad config "consentHandlingOverride: true/false". it is false by default
- any ad network integrated with amp-consent can flip the bit, and then they will have the "current behavior"
- any ad network has "consentHandlingOverride: false" will be blocked completely unless user accepts consent.

### PR DESCRIPTION
**Pull request by lannka**
Closes #15269

### CODE REVIEW

File: 3p/3p.js [hunk 1]
```diff
@@ -272,6 +272,7 @@ function validateAllowedFields(data, allowedFields) {
     location: true,
     mode: true,
     consentNotificationId: true,
+    consentHandlingOverride: true,
```
Comment (zhouyx) [inline 3p/3p.js]: this should be `blockOnConsent`, the attribute is `data-block-on-consent`

File: ads/_config.js [hunk 2]
```diff
@@ -160,6 +161,7 @@ export const adConfig = {
     remoteHTMLDisabled: true,
     masterFrameAccessibleType: 'google_network',
     fullWidthHeightRatio: 1.2,
+    consentHandlingOverride: true,
```
[the remark this answers is not in the document]
Comment (zhouyx) [reply ads/_config.js]: and here

File: 3p/3p.js (hunk 1 above)
Comment (lannka) [reply 3p/3p.js]: done

Comment (lannka) [reply 3p/3p.js]: done

File: ads/_config.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lannka) [reply ads/_config.js]: done

File: 3p/3p.js (hunk 1 above)
Comment (zhouyx) [reply 3p/3p.js]: I couldn't see the name change : )

Comment (lannka) [reply 3p/3p.js]: woops. really done now. thanks!

**Pull request merged**